streams/src/main/java/org/apache/kafka/streams/state/internals/MeteredVersionedKeyValueStore.java

```diff
@@ -84,22 +86,37 @@
         extends MeteredKeyValueStore<K, ValueAndTimestamp<V>> {
 
         private final VersionedBytesStore inner;
+        private final Serde<V> rawValueSerde;
+        private StateSerdes<K, V> rawValueSerdes;
 
         MeteredVersionedKeyValueStoreInternal(final VersionedBytesStore inner,
                                               final String metricScope,
                                               final Time time,
                                               final Serde<K> keySerde,
-                                              final Serde<ValueAndTimestamp<V>> valueSerde) {
-            super(inner, metricScope, time, keySerde, valueSerde);
+                                              final Serde<V> valueSerde) {
+            super(
+                inner,
+                metricScope,
+                time,
+                keySerde,
+                valueSerde == null
+                    ? null
+                    : new ValueAndTimestampSerde<>(valueSerde)
+            );
             this.inner = inner;
+            this.rawValueSerde = valueSerde;
         }
 
-        @Override
-        public void put(final K key, final ValueAndTimestamp<V> value) {
-            if (value == null) {
-                throw new IllegalStateException("Versioned store requires timestamp associated with all puts, including tombstones/deletes");
+        public boolean put(final K key, final V value, final long timestamp) {
+            Objects.requireNonNull(key, "key cannot be null");
+            try {
+                final boolean isLatest = maybeMeasureLatency(() -> inner.put(keyBytes(key), rawValueSerdes.rawValue(value), timestamp), time, putSensor);
```

Comment:
I am confused. I don't see a change to `StreamsMetricsImpl#maybeMeasureLatency` to return a boolean? Can we even use a `Runable` as we now need a return type?

Reply:
There's another existing implementation which takes `Supplier<T>` and passes the result through: https://github.com/apache/kafka/blob/f252c75bf3f2db93bf062bc7e49d3c79a06e9568/streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImpl.java#L876

It's the same one that's currently being used in the file to return `ValueAndTimestamp<V>` from `get(key)` in the method below this one.

Reply:
Ah. Missed this one. Nice. Was really wondering as your PR compiled :)